display/run.sh

```diff
@@ -2,30 +2,46 @@
 
 set -e
 
-# SINK.tally params
+# SINK.{tally,print} params
 display_compact=true
 
 # Generate interval messages "interval_callbacks.c"
 ruby ./source_callbacks_generator.rb 3.interval_instances.yaml SOURCE.interval/interval_callbacks.c
 
 # Generate SOURCE.invertal component
-ruby ../main.rb -d 2.interval_definitions.yaml -t SOURCE -p convert --params 1.params.yaml -c interval
+ruby ../main.rb -d 2.interval_definitions.yaml -t SOURCE -p sample -c interval
 make -C SOURCE.interval
 babeltrace2 --plugin-path=SOURCE.interval \
-            --component=source.convert.interval \
+            --component=source.sample.interval \
             --component=sink.text.details 
 
 # Genarate SINK.tally component
 ruby ../main.rb -u 2.interval_definitions.yaml -t SINK -p display --params 1.params.yaml -c tally
 make -C SINK.tally
 babeltrace2 --plugin-path=SOURCE.interval:SINK.tally  \
-            --component=source.convert.interval \
+            --component=source.sample.interval \
             --component=sink.display.tally  \
             --params="display_compact=${display_compact}"  
 
-rm -f component.h dispatch.{h,c,o} params.{h,c,o} tally.{c,o,so} my_callbacks.o my_demangle.o
+
+# Genarate FILTER.aggreg component
+ruby ../main.rb -u 2.interval_definitions.yaml -d 4.aggreg_definitions.yaml -t FILTER -p by_backend --params 1.params.yaml -c aggreg
+make -C FILTER.aggreg
+babeltrace2 --plugin-path=SOURCE.interval:FILTER.aggreg  \
+            --component=source.sample.interval \
+            --component=filter.by_backend.aggreg
+
+# Genarate SINK.print component
```

Comment:
Please don't call it print, it's to general. Call it `sink.tally.print` 
and the other one `filter.tally.aggregation`

Reply:
We can not create two separate components with the same plugin name, since it failed. That's why I did not do it. I think is because the components of the same plugging should be declared in the same babeltrace package. I did not go into details.

```bash
--component=filter.tally.aggreg \
--component=sink.tally.print
```

Reply:
Oh, it will be usefull if we can make it work; I think it's clearer. But not super important